site-src/geps/gep-1762.md

```diff
@@ -0,0 +1,309 @@
+    protocol: HTTP
+```
+
+With this configuration, an implementation:
+* MUST treat the `parent-gateway` as having the concatenated list of all listeners from merged Gateways.
+  This means, for example, that a Service with port 80 and 81 may be created in this example.
+
+This introduces a new concept in Gateway merging of a "primary" Gateway.
+This makes it explicit which is the primary gateway and which are attached to it.
+
+Secondary Gateways cannot configure `addresses` or anything in `infrastructure` (aside from `attachTo`), and must have
+the same `gatewayClassName` as the primary Gateway they are attached to.
+
+Gateways can only `attachTo` a primary Gateway. That is, we cannot have a chain of references.
+
+### Customizations
+
+With any in-cluster deployment, customization requirements will arise. 
+
+Some common requirements would be:
+* `Service.spec.type`, to control whether a service is a `ClusterIP` or `LoadBalancer`.
+* IP in the Service to assign to it.
+* Arbitrary labels and annotations on generated resources.
+* Any other arbitrary fields; the list is unbounded. Some examples would be:
+  * CPU and memory requests
+  * Service `externalTrafficPolicy`
+  * Affinity rules.
+
+#### Gateway Type
+
+This is handled by [GEP-1651](https://github.com/kubernetes-sigs/gateway-api/pull/1653), so won't be described here.
+
+#### Gateway IP
+
+For setting a specific IP address in a generated `Service`, the `.Spec.Address` field can be used.
+
+For example:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: my-gateway
+spec:
+  addresses:
+  - type: IPAddress
+    value: 1.1.1.1
+  gatewayClassName: example
+  listeners:
+  - name: default
+    port: 80
+    protocol: HTTP
+```
+
+This would generate a `Service` with `clusterIP` or `loadBalancerIP`, depending on the Service type.
+
+Note: this differs from the selection of "manual deployment", which doesn't use `IPAddress` type.
+
+#### Labels and Annotations
+
+Labels and annotations SHOULD be copied from the `Gateway` to generated resources.
+
+Note this may mean an annotation intended for a `Service` may end up on a `Deployment` (for example).
+This is typically not a concern.
+However, if an implementation is aware of specific meanings of certain labels or annotations, they MAY
+exclude these from irrelevant resources.
+
+#### Arbitrary Customization
+
+Currently, to provide arbitrary customizations, Gateway API provides a few mechanisms:
+
+* GatewayClass has a `parametersRef` field, allowing to point to an arbitrary other resource.
+
+  This *can* be used to make arbitrary customizations, but comes with a number of issues:
+  * GatewayClass is cluster scoped. This means that any user configuring a namespace-scoped Gateway
+    may need to make a new cluster-scoped resources, which makes RBAC extremely challenging.
+    An analogy would be to have a `DeploymentClass` to configure CPU requests, instead of a field in each `Deployment`.
```

Comment:
Well this is fair, we do have many workload resources that are used to provide different ways to schedule Pods (Deployment, StatefulSet, DaemonSet, etc). With the resource class model, it would be reasonable to say that these concepts would map to different Gateway Classes. Of course my reference example here is Internal vs External LBs which often represent different kinds of underlying infra. So while I agree that we probably don't want to represent something like Resource Requests with a separate GWC, I don't think we want to discount the use of GWC entirely.